### ISSUE TITLE
Re-migrate globals not working as expected when upgrading from Elementor 2.x to 3.x

### ISSUE THREAD

**Issue by jvernooy** [elementor/elementor#15477]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

When upgrading from Elementor 2.x to 3.x, the re-migrate globals functionality does not seem to work correctly. Theme Styles, Fonts and other global settings are not being migrated correctly. Triggering the upgrade script again via _Tools -> Version Control -> Rerun Update Script_ causes the following error, resulting in the request failing with status 500 - server error:

`PHP Fatal error:  Uncaught Error: Cannot unset string offsets in /var/www/html/wp-content/plugins/elementor/core/upgrade/elementor-3-re-migrate-globals.php:77`

It seems like the upgrade script looks up the PostMeta value of `_elementor_page_settings` for the "Default Kit" post. When it does not exist, the lookup function returns a blank string value, but the rest of the upgrade logic expects the value to be an array. While processing the various style items that need to be migrated, the code attempts to unset specific entries from the array, which fails since calling `unset` on a string is invalid.

## Steps to reproduce

1. Set up a new site with Elementor 2.x (we used 2.19.14 with the Pro plugin at 2.10.3)
2. Set up a custom font, change the default page width, and apply a Theme Style background to all posts.
3. Ensure all other plugins are disabled, and that the default WordPress theme is active.
4. Upgrade the Elementor plugin to the latest stable version (currently 3.2.5 according to the WordPress plugins page)
5.  Go ahead and try to run the "Rerun Update Scripts" functionality from _Tools -> Version Control_

## Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>

```
        == Server Environment ==
	        Operating System: Linux
	        Software: nginx/1.17.5
	        MySQL version: Source distribution v8.0.20
	        PHP Version: 7.4.13
	        PHP Max Input Vars: 1000
	        PHP Max Post Size: 200M
	        GD Installed: Yes
	        ZIP Installed: Yes
	        Write Permissions: All right
	        Elementor Library: Connected
        
        == WordPress Environment ==
	        Version: 5.7.2
	        Site URL: <some internal company url>
	        Home URL: <some internal company url>
	        WP Multisite: Yes
	        Max Upload Size: 73 MB
	        Memory limit: 512M
	        Permalink Structure: /%postname%/
	        Language: en-US
	        Timezone: 0
	        Debug Mode: Inactive
        
        == Theme ==
	        Name: 
	        Version: 
	        Author: Kurtosys Systems
	        Child Theme: No
        
        == User ==
	        Role: administrator
	        WP Profile lang: en_US
	        User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:89.0) Gecko/20100101 Firefox/89.0
        
        == Network Plugins ==
	        Elementor
		        Version: 3.2.3
		        Author: Elementor.com
        
	        Elementor Pro
		        Version: 3.2.2
		        Author: Elementor.com
        
        
        == Elementor - Compatibility Tag ==
	        
        
        
        == Elementor Pro - Compatibility Tag ==
	        
        
        
        == Elements Usage ==
	        
	        header : 22
		        button : 17
		        heading : 17
		        html : 1
		        icon : 11
		        image : 2
		        nav-menu : 1
		        text-editor : 1
	        page : 211
		        blockquote : 2
		        button : 6
		        divider : 6
		        form : 1
		        heading : 13
		        icon : 2
		        icon-list : 1
		        image : 8
		        image-box : 3
		        text-editor : 12
		        video : 1
	        popup : 50
		        form : 1
		        heading : 1
	        section : 636
		        divider : 2
		        form : 27
		        heading : 4
		        image : 3
		        text-editor : 6
	        widget : 0
		        expandable-text : 1
        
        == Log ==

```
</details>

**Comment by Aviad-Herman** [comment]
Thanks for the investigation, I will ask dev tomorrow.
BTW, some people with similar issues have reported that temporarily deactivating other plugins (specifically SEOPress plugin and / or OceanWP Theme) solved the issue for them so you might want to try that too meanwhile.

**Comment by shilo-ey** [comment]
@jvernooy Please see comments on the PR.

Thanks!

**Comment by shilo-ey** [comment]
This issue has been resolved in Elementor v3.4.0

Feel free to update

Thanks!
